### PR TITLE
Adding VPC Config capability & removing env var check from 'run'

### PR DESCRIPTION
**Pull request by scottjpack**
Adding a few new things:

@dan-v These probably don't fit what you're using it for, but you just laid such awesome groundwork!  If you don't want these messing up the repo I can maintain them in a separate fork.

1 - Can now deploy lambda functions with VPCConfig:
If "run" launched with the -s (subnet ID) and -g (security group) flags, the lambda function will be deployed with a VPCConfig.  The "AWSLambdaVPCAccessExecutionRole" managed policy is now attached to the Execution Role as part of setup to facilitate this.  If the flags are not included, it will create the function with no VPCConfig.  *Note: If the VPC the function is launched into does not have a NAT Gateway it will not have access to the internet, preventing the reverse ssh tunnel from working. Which brings us to....

2 - Can now specify the destination of the SSH target.
If "run" is launched with the -i option the parameter will be used in place of the public IP of the host running the awslambdaproxy.  This allows the tunnel to be set up using a VPC Endpoint Service.

3 - Can now use instance roles.
I removed the checks for environment variables from run.go.  This allows the SDK to use whatever permissions are set up on this host, including a .aws/credentials file or an instance role.

### CODE REVIEW

**Comment by scottjpack** [comment]
Ooh, scratching this.  Auth via role needs more testing.

**Comment by dan-v** [comment]
@scottjpack - Thanks for pull request (even if you closed it haha). My primary motivation for using non VPC setup is to get that giant pool of IP addresses, so I'm curious what your use case is with using VPC.

**Comment by scottjpack** [comment]
@dan-v I'm trying to get network connectivity to instances in VPCs in other accounts (where I have an IAM role) without modifying security groups, routing rules, or creating instances. I'm hoping to use it for vulnerability scanning on segmented/isolated networks.

**Comment by scottjpack** [comment]
NM, it's fine.  Thought I was missing some EC2 read permissions, but they were included by the managed policy.

**Comment by dan-v** [comment]
@scottjpack - that's actually a really cool use case. Let me ponder if it makes sense to pull these changes in. I'll take a look at the code changes as well.

File: infrastructure.go [hunk 1]
```diff
@@ -3,277 +3,330 @@ package awslambdaproxy
...
+                                if awsErr.Code() == "InvalidParameterValueException" {
+                                        time.Sleep(time.Second)
+                                        return infra.createLambdaFunction(svc, roleArn, payload)
+                                }
+                        }
+                        return err
+                }
+                return createLambdaAlias(svc, function.Version)
+        }
+        if infra.subnetId != "" {
```
Comment (dan-v) [inline infrastructure.go:196]: A lot of duplicate code here. Should find a way to just optionally pass in subnet/sg.

File: init.go [hunk 2]
```diff
@@ -7,15 +7,20 @@ import (
 )
 
 // ServerInit is the main entrypoint for the server portion of awslambdaproxy
-func ServerInit(sshUser string, sshPort string, regions []string, memory int64, frequency time.Duration,
-	listeners []string, timeout int64) {
+func ServerInit(sshUser string, sshPort string, publicIp string, regions []string, memory int64, frequency time.Duration,
+	listeners []string, timeout int64, subnetId string, sgId string) {
+
 	publicIP, err := getPublicIP()
```
Comment (dan-v) [inline init.go:13]: I'd just skip calling this function if publicIP is provided, rather than overriding it after the fact.

File: run.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
+package cmd
```
Comment (dan-v) [inline run.go]: Hmm.. not sure how this file got to the root location. This should be in cmd/awslambdaproxy/cmd/run.go

File: run.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+			os.Exit(1)
+		}
+
+		// check frequency
+		if aFrequency > lambdaMaxFrequency {
+			fmt.Println("Maximum lambda frequency is " + lambdaMaxFrequency.String() + " seconds")
+			os.Exit(1)
+		}
+
+		// check for required aws keys
```
Comment (dan-v) [inline run.go]: I definitely like the idea of allowing the SDK to find credentials, but I think this approach will currently fail if no valid credentials are available. A better approach might be to run a simple command to check valid credentials exist before doing anything else.

**Comment by scottjpack** [comment]
Thanks for reviewing it, sorry it is pretty messy.  I'm coming from Python,
and am still figuring out how variable scoping/assignment works in Go.

**Comment by dan-v** [comment]
Gonna close this one out. Not going to support this use case for now. Hope that it is working well for you though!

**Pull request closed without merging**